### PR TITLE
Support Windows in incremental_build.sh

### PR DESCRIPTION
**Pull request by stuartmorgan**
## Description

Allow running incremental_build.sh on Windows under Git Bash. Longer
term this script should be re-written in Dart (#64025), but for now this
allows running the script on Windows bots.

## Related Issues

https://github.com/flutter/flutter/issues/56143

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: script/incremental_build.sh [hunk 1]
```diff
@@ -6,6 +6,12 @@ readonly REPO_DIR="$(dirname "$SCRIPT_DIR")"
 
 source "$SCRIPT_DIR/common.sh"
 
+if [ "$(expr substr $(uname -s) 1 5)" == "MINGW" ]; then
+  PUB=pub.bat
+else
+  PUB=pub
+fi
```
Comment (dnfield) [inline script/incremental_build.sh:13]: Could the if here just be `command -v pub.bat`?

Comment (stuartmorgan) [reply script/incremental_build.sh:13]: What's the advantage of that approach? If for some reason there were an executable called `pub.bat` in the path on Linux or macOS we wouldn't want to run it, and if `pub.bat` isn't there on Windows `pub` still wouldn't be the right command to try to run, so it seems like an odd construction to me.

Comment (dnfield) [reply script/incremental_build.sh:13]: I'm worried if the Windows image changes this will break. But that should be covered by CI.  LGTM.

Comment (stuartmorgan) [reply script/incremental_build.sh:13]: Hopefully we'll rewrite it in Dart before then :)

Comment (dnfield) [reply script/incremental_build.sh:13]: You are much more optimistic than I am :D

**Pull request merged**